Join tutorial: Show up-leveling of fields from embedded record (#4492)
brimdata/zed: docs/tutorials/join.md

{name:"apple",color:"red",flavor:"tart",eaterinfo:{name:"morgan",age:61,likes:"tart"}}
{name:"apple",color:"red",flavor:"tart",eaterinfo:{name:"chris",age:47,likes:"tart"}}
```

> If having the data in an embedded record is undesirable, it can easily be
> copied to another location, such as with the
> [spread operator](../language/overview.md#7112-record-expressions). Additional

Reviewer: Maybe state the goal here more directly?
```suggestion
If embedding the opposite record is undesirable, the left and right
records can easily be merged with the
[spread operator](../language/overview.md#7112-record-expressions). Additional
```

processing may be necessary to handle conflicting field names, such as
in the example just shown where the `name` field is used differently in the

> left and right data sources. We can account for this by augmenting our Zed into
> a new script `upleveled-opposite.zed`.

Reviewer: 1. We mostly say "input" in this file.
2. Rather than "uplevel", use whatever term we choose to describe our goal here.
3. "augment our Zed into" reads oddly.
```suggestion
left and right inputs. We'll demonstrate this by augmenting `embed-opposite.zed`
to produce `merge-opposite.zed`.
```

```mdtest-input upleveled-opposite.zed
from (
  file fruit.ndjson => sort flavor
  file people.ndjson => sort likes

> ) | inner join on flavor=likes eaterinfo:=this

Author: @nwt: Oops. Thanks for catching. Indeed, I was modifying an example in a directory that had Zed scripts from a prior iteration still hanging around.
